qiskit/visualization/matplotlib.py

```diff
@@ -358,133 +363,137 @@ def _get_gate_ctrl_text(self, op):
 
     def _get_colors(self, op):
         base_name = None if not hasattr(op.op, 'base_gate') else op.op.base_gate.name
-        if op.name in self._style.dispcol:
-            color = self._style.dispcol[op.name]
+        if op.name in self._style['dispcol']:
+            color = self._style['dispcol'][op.name]
             # Backward compatibility for style dict using 'displaycolor' with
             # gate color and no text color, so test for str first
             if isinstance(color, str):
                 fc = color
-                gt = self._style.gt
+                gt = self._style['gt']
             else:
                 fc = color[0]
                 gt = color[1]
         # Treat special case of classical gates in iqx style by making all
         # controlled gates of x, dcx, and swap the classical gate color
-        elif self._style.name == 'iqx' and base_name in ['x', 'dcx', 'swap']:
-            color = self._style.dispcol[base_name]
+        elif self._style['name'] == 'iqx' and base_name in ['x', 'dcx', 'swap']:
+            color = self._style['dispcol'][base_name]
```

Comment:
In general, are these dictkeys mandatories?

Reply:
could it be that, for example `self._style['dispcol']` does not exists?

Reply:
Oh, I see. `self._style = DefaultStyle().style` is loaded at the top, which includes 'dispcol' and after that calls to 'set_style' can only change the values of items, not remove them. So I think it's safe.